test/parallel/test-fs-chmod-mask.js

```diff
@@ -89,6 +90,34 @@ function test(mode, asString) {
     fs.lchmodSync(link, input);
     assert.strictEqual(fs.lstatSync(link).mode & 0o777, mode);
   }
+
+  // Test that `lchmod` works on readonly (0x444) files.
+  if (fs.lchmod) {
+    const link = path.join(tmpdir.path, `lchmod-read-only-src-${suffix}`);
+    const file = path.join(tmpdir.path, `lchmod-read-only-dest-${suffix}`);
+    fs.writeFileSync(file, 'test', { encoding: 'utf-8', mode: 0o444 });
+    fs.symlinkSync(file, link);
+
+    fs.lchmod(link, input, common.mustCall((err) => {
+      assert.ifError(err);
+      assert.strictEqual(
+        fs.lstatSync(link).mode & kMaskOnlyRead, kMaskOnlyRead
```

Comment:
`fs.lstatSync(link).mode, mode` is not going to work - the former contains both file type and permission info (so `> 0o777`) and the latter is `0o777` on non-windows systems in this test (see the top of this file).